src/index.ts

```diff
@@ -103,13 +104,13 @@ export const createInstance = (config: Config): Instance => {
       lockStatus: boolean,
       signer: ethers.Signer
     ): Promise<ethers.ContractTransaction> => {
-      const registrar: Registrar = await getRegistrar(signer, config.registrar);
+      const hub: ZNSHub = await getHubContract(signer, config.hub);
```

Comment:
I suppose you could simplify this even further by just having `config.hub` be a parameter to the action, and calling `getHubContract` inside the action, same for the other functions

Reply:
Not sure what the best pattern will be here yet, I think we'll need to do a refactor pass in the future after the launch.

Making the actions just take a registrar and then doing the domainId->registrar look up outside of them may be wiser.